diff --git a/src/index.ts b/src/index.ts
index 3fc53118b24a10975284d24a8fc82b1d1fbd799e..21ade36ae99d1f7aab902da9fbfb9dc909fbaa3a 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -60,7 +60,12 @@ export const logger = (options?: Options) => {
                 logStr.push(String(error.status))
             }
 
-            logStr.push(error.message)
+            const errorMessage =
+                'message' in error ? error.message :
+                'value' in error ? String(error.value) :
+                'Error details not available';
+
+            logStr.push(errorMessage)
             const beforeTime: bigint = (store as any).beforeTime;
 
             logStr.push(durationString(beforeTime))
